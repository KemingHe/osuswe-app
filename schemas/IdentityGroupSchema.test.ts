import {
  IDENTITY_GROUP_OPTIONS,
  IdentityGroupSchema,
} from '@/schemas/IdentityGroupSchema';

describe('IDENTITY_GROUP_OPTIONS', () => {
  it('matches the latest snapshot', () => {
    expect(IDENTITY_GROUP_OPTIONS).toMatchSnapshot();
  });
});

describe('IdentityGroupSchema', () => {
  it('parses valid identity groups', () => {
    expect(IdentityGroupSchema.parse('African-American')).toBe(
      'African-American',
    );
    expect(IdentityGroupSchema.parse('Asian Connection')).toBe(
      'Asian Connection',
    );
    expect(
      IdentityGroupSchema.parse(
        'HeForSWE (male allies of women in engineering)',
      ),
    ).toBe('HeForSWE (male allies of women in engineering)');
    expect(IdentityGroupSchema.parse('Latinx')).toBe('Latinx');
    expect(IdentityGroupSchema.parse('LGBTQ+')).toBe('LGBTQ+');
    expect(
      IdentityGroupSchema.parse('Native-American/Indigenous Peoples'),
    ).toBe('Native-American/Indigenous Peoples');
    expect(IdentityGroupSchema.parse('Military and Veterans')).toBe(
      'Military and Veterans',
    );
    expect(IdentityGroupSchema.parse('Disability Inclusion')).toBe(
      'Disability Inclusion',
    );
    expect(
      IdentityGroupSchema.parse(
        'I do not identify with any of the above groups',
      ),
    ).toBe('I do not identify with any of the above groups');
  });

  it('throws an error for invalid identity groups', () => {
    expect(() => IdentityGroupSchema.parse('invalid')).toThrow();
    expect(() => IdentityGroupSchema.parse('')).toThrow();
  });

  it('throws an error for invalid types', () => {
    expect(() => IdentityGroupSchema.parse(123)).toThrow();
    expect(() => IdentityGroupSchema.parse({})).toThrow();
    expect(() => IdentityGroupSchema.parse([])).toThrow();
    expect(() => IdentityGroupSchema.parse(null)).toThrow();
    expect(() => IdentityGroupSchema.parse(undefined)).toThrow();
  });

  it('throws the correct error message for invalid string', () => {
    const result = IdentityGroupSchema.safeParse('invalid');
    expect(result.success).toBe(false);
    expect(result.error?.issues[0].message).toBe(
      'Invalid identity group selection',
    );
  });

  it('throws the correct error message for valid type', () => {
    const result = IdentityGroupSchema.safeParse(123);
    expect(result.success).toBe(false);
    expect(result.error?.issues[0].message).toBe(
      'Invalid identity group selection',
    );
  });
});
