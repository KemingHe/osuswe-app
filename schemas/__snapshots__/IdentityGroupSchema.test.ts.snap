// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IDENTITY_GROUP_OPTIONS > matches the latest snapshot 1`] = `
[
  "African-American",
  "Asian Connection",
  "HeForSWE (male allies of women in engineering)",
  "Latinx",
  "LGBTQ+",
  "Native-American/Indigenous Peoples",
  "Military and Veterans",
  "Disability Inclusion",
  "I do not identify with any of the above groups",
]
`;
